### PR TITLE
perf(template-compiler): static-optimize `spellcheck`

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

Applies the static content optimization to elements with the `spellcheck` attribute.

We handle the `spellcheck` attribute in a special way. This seems due to a bug from years ago: https://github.com/salesforce/lwc/issues/1595

In any case, this PR makes it so static-optimized nodes render `spellcheck` in exactly the same way as non-static-optimized nodes. There are several tests to prove this.

Partially addresses #4280

## Does this pull request introduce a breaking change?

-   😮‍💨 No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

-   🤞 No, it does not introduce an observable change.

### CODE REVIEW

File: packages/@lwc/integration-karma/test/component/spellcheck-attribute/index.spec.js [hunk 1]
```diff
@@ -49,6 +49,7 @@ describe('setting static values on non custom elements', () => {
         const elm = createElement('x-container', { is: Container });
         document.body.appendChild(elm);
 
+        expect(getSpellcheckValue(elm, 'div.boolean-true-v')).toBe('');
```
Comment (nolanlawson) [inline packages/@lwc/integration-karma/test/component/spellcheck-attribute/index.spec.js:52]: This test was missing in the original test! I added it for completeness.

There are 8 cases that are being tested here:

https://github.com/salesforce/lwc/blob/7b0418c79e4159c5a8d0f8bda5ded0f1cf312ce7/packages/%40lwc/integration-karma/test/component/spellcheck-attribute/x/container/container.html#L11-L18

File: packages/@lwc/template-compiler/src/__tests__/fixtures/attributes/spellcheck/expression/expected.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import _implicitStylesheets from "./expression.css";
 import _implicitScopedStylesheets from "./expression.scoped.css?scoped=true";
 import _xFoo from "x/foo";
 import { freezeTemplate, parseFragment, registerTemplate } from "lwc";
-const $fragment1 = parseFragment`<div${"a0:spellcheck"}${3}></div>`;
+const $fragment1 = parseFragment`<div${"true"}${3}></div>`;
```
Comment (nolanlawson) [inline packages/@lwc/template-compiler/src/__tests__/fixtures/attributes/spellcheck/expression/expected.js:5]: OK this is wrong... will need to circle back on this.

Comment (nolanlawson) [reply packages/@lwc/template-compiler/src/__tests__/fixtures/attributes/spellcheck/expression/expected.js]: Fixed edfbb26c2

File: packages/@lwc/integration-karma/test/component/spellcheck-attribute/index.spec.js [hunk 3]
```diff
@@ -71,3 +74,41 @@ describe('setting static values on non custom elements', () => {
...
+            document.body.appendChild(elm);
+
+            elm.theSpellcheck = setValue;
+
+            await Promise.resolve();
+            expect(getSpellcheckValue(elm, '.dynamic')).toBe(expectedAttrValue);
+            expect(getSpellcheckPropertyValue(elm, '.dynamic')).toBe(expectedPropValue);
+        });
+    }
+});
```
Comment (nolanlawson) [inline packages/@lwc/integration-karma/test/component/spellcheck-attribute/index.spec.js:114]: Was worried about regressing dynamically updating the `spellcheck` value, so added a bunch of tests.

These tests are just asserting that our static behavior matches our non-static (historical) behavior.

**Review by wjhsf — APPROVED** [review]

**Pull request merged**